File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n=nums.length;
        int low=0;
        int mid=0;
        int high=n-1;
        int temp=0;
        while(mid<=high){
            if(nums[mid]==0){
                temp=nums[low];
                nums[low]=nums[mid];
                nums[mid]=temp;
                low++;
                mid++;
            }
           else if(nums[mid]==1){
                mid++;
            }
           else if(nums[mid]==2){
                temp=nums[high];
                nums[high]=nums[mid];
                nums[mid]=temp;
                high--;
            }
            
            }
        }
    }

File: 0075-sort-colors/0075-sort-colors.cs
public class Solution {
    public void SortColors(int[] nums) {
        int n=nums.Length;
        int k=0;
        Dictionary<int,int> dict=new Dictionary<int,int>();
        for(int i=0;i<n;i++){
            if(!dict.ContainsKey(nums[i])){
                dict.Add(nums[i],1);
            }
            else{
                dict[nums[i]]++;
            }
        }
        var myList=dict.OrderBy(x=>x.Key).ToList();
        int [] res=new int[n];
        foreach(var item in myList){
            for(int i=0;i<item.Value;i++){
                res[k]=item.Key;
                k++;
            }
        }
        for(int j=0;j<res.Length;j++){
            nums[j]=res[j];
        }
        
    }
}

